### ISSUE TITLE
UnusedPrivateMember - [invoke]: False positive: Used by public member function

### ISSUE THREAD

**Issue by hfhbd** [detekt/detekt#4435]
## Expected Behavior
no issue

## Observed Behavior
`UnusedPrivateMember - [invoke] at /Users/philipwedemann/invokeFalsePositive/src/main/kotlin/Test.kt:2:5`

## Steps to Reproduce
`./gradlew detekt`

```kt
object Test {
    private operator fun invoke(i: Int): Int = i

    fun answer() = Test(1)
}

val answer = Test.answer()
```
See also the attached reproducer project
## Context
I adopt the code style from kotlinx.coroutines, which uses this constructor like function to emulate a constructor for an interface:
```kt
interface MutableStateFlow { }
fun MutableStateFlow(initValue: t) { }
```
Using this code style for the `invoke` function and combine it with `object`s results into some kind of namespaces.
## Your Environment

* Version of detekt used: 1.19
* Version of Gradle used: 7.3.3
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/rhwzv6cdovkwe
* Operating System and version: macOS 12.1
* Link to your project (if it's a public repository):
[invokeFalsePositive.zip](https://github.com/detekt/detekt/files/7799166/invokeFalsePositive.zip)

**Comment by BraisGabin** [comment]
Related with #4242.

**Comment by BraisGabin** [comment]
@hfhbd your issue I think that it will be solved by #4441 because you aren't using type solving. But I'll keep this open because we have the same issue with type solving too.

**Comment by chao2zhang** [comment]
@BraisGabin Is the stance that we should acknowledge such degradation of `UnusedPrivateMember` with type resolution?

**Comment by atulgpt** [comment]
Hi @BraisGabin, this is also blocked for my project. I have created a PR for the fix at https://github.com/detekt/detekt/pull/5717
